### PR TITLE
Slightly improved

### PR DESCRIPTION
**Pull request by angerman**
This makes the whole thing a bit more configurable.

We should probably read the config from a file? (e.g. a `settings` file, like the one ghc has?)

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -13,19 +13,42 @@ import qualified Data.Set as Set
...
 main =  do
       hSetBuffering stdout LineBuffering
-      shelly $ verbosely $ do
+      shelly $ verbosely $ mkOverlay (Config
+                                      "patches"
+                                      ".keys"
+                                      ".tmpl"
+                                      "cabal.cfg"
+                                      "packages"
+                                      "hackage.haskell.org"
```
Comment (hvr) [inline src/Main.hs]: Please make these configurable via CLI (e.g. optparse-applciative), as they're obviously specific to the installation (i.e. gtar/urls). If you want, you can make the foldernames configurable as well, but it's not that essential (as one can symlink as workaround).

**Review by hvr — CHANGES_REQUESTED** [review]

**Comment by angerman** [comment]
This now includes a fix for #3.

**Comment by angerman** [comment]
I believe this is now ready to merge. Let me know what you think @hvr.

File: src/Main.hs (hunk 1 above)
Comment (angerman) [reply src/Main.hs]: I think this is resolved now?

File: tool.cabal [hunk 2]
```diff
@@ -21,6 +21,7 @@ executable tool
                , text            ^>= 1.2.2.2
                , system-filepath ^>= 0.4.13.4
                , containers      ^>= 0.5.6.2
+               , optparse-applicative
```
Comment (hvr) [inline tool.cabal]: ...I think you forgot a version annotation here... :-)

**Review by hvr — CHANGES_REQUESTED** [review]

**Comment by hvr** [comment]
thanks!

**Pull request merged**